pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -56,6 +67,11 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 		return nil
 	}
 
+	err := r.reconcileTlsSecret(ctx)
```

Comment:
Shouldn't this be done after other checks like `if r.dk.Status.KubeSystemUUID == "" `?
Logic: why create a secret if the status is not even ready so we will fail either way

Reply:
also will this secret be cleaned-up in case `Extensions` get disabled?

this leads me to: 
wouldn't it be cleaner to have this secret reconciliation in a separate reconciler that is called somewhere at: https://github.com/Dynatrace/dynatrace-operator/blob/da71084911b1db6fab4d02ab1ca1e3b71507e4fe/pkg/controllers/dynakube/extension/reconciler.go#L47 ?

you can ignore my first comment of 
> why create a secret if the status is not even ready so we will fail either way
I thought this reconciler is the "main" `Extensions` reconciler, but this is only for the statefulset, so IMO the secret reconcile shouldn't be here

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/fd1af2ce0cfdf8f64e04761bf426598e57afa58e)
Adding more unit tests now